pkg/querier/querier.go

```diff
@@ -119,17 +119,17 @@ func (q *V3ioQuerier) selectQry(
 			sets[i] = set
 		}
 
-		// Sort each partition when not using range scan
-		if name == "" {
-			for i := 0; i < len(sets); i++ {
-				// TODO make it a Go routine per part
-				sorter, err := NewSetSorter(sets[i])
-				if err != nil {
-					set = nullSeriesSet{}
-					return
-				}
-				sets[i] = sorter
+		// Sort each partition
+		// todo: Removed condition that applies sorting only on non range scan queries, to fix bug with serieses coming OOO when querying multi partitions.
+		// todo: Think of a better solution.
+		for i := 0; i < len(sets); i++ {
+			// TODO make it a Go routine per part
+			sorter, err := NewSetSorter(sets[i])
+			if err != nil {
+				set = nullSeriesSet{}
```

Comment:
log the errors
Consider adding local logger as `methodLogger := q.logger.GetChild("selectQry")` and then using it within the method for logging necessary stuff. Here for example you can do: `methodLogger.Error("Error: %v", err)`

Reply:
Note that this method has named return types. so the error is returned and it's up to the caller to manage the error. it gets logged there

Reply:
Just make sure that we don't miss those errors in the log when running from Prom UI or REST